[TNT-181] 트레이니 홈 화면 일별 기록 UI 구현

SeonJeongk opened the pull request:
## 📝 작업 내용

- #44 

- 트레이니 홈 화면 일별 기록 UI 구현했습니다
- API 연결하면서 DateFormatter로 시간 변환하겠습니다. 


## 📸 실행 화면
https://github.com/user-attachments/assets/fe5651a2-d042-4c8e-945d-1e1593557b2a





## 🙆🏻 리뷰 요청 사항
없습니다아


## 👀 레퍼런스

SeonJeongk committed:
[TNT-181] refactor: RecordChip에 ChipStyle 추가

SeonJeongk committed:
[TNT-181] feat: 선택된 날짜의 PT수업과 기록 UI 구현

SeonJeongk committed:
[TNT-181] fix: 기록/일정 있는 날짜에 icon 보이도록 수정

SeonJeongk committed:
[TNT-181] fix: 배경색 수정

SeonJeongk committed:
[TNT-181] fix: lint 오류 해결

SeonJeongk committed:
[TNT-181] refactor: Scaffold 대신 LazyColumn으로 구현

SeonJeongk committed:
[TNT-181] feat: 트레이니 홈화면 기록 관련 data class 구현

SeonJeongk committed:
[TNT-181] feat: TraineeRepository 생성

SeonJeongk committed:
[TNT-181] fix: Repository에서 UI 데이터 받아오도록 수정

SeonJeongk committed:
[TNT-181] refactor: 변수명 수정

SeonJeongk committed:
[TNT-181] fix: 선택된 날짜 관리 state로 통일

SeonJeongk committed:
[TNT-181] fix: import 순서 변경

SeonJeongk committed:
[TNT-181] refactor: composable 분리

SeonJeongk committed:
[TNT-181] refactor: 화면 간격 수정

hoyahozz commented on feature/trainee/home/src/main/java/co/kr/tnt/trainee/home/TraineeHomeScreen.kt:
```diff
@@ -62,82 +95,285 @@ internal fun TraineeHomeRoute(
+                )
+            }
+        }
+        if (state.recordList.isNullOrEmpty()) {
```
`null` 과 `empty` 가 같은 의미인지 한 번 더 생각해보면 좋을 것 같습니다 ~!

hoyahozz commented on feature/trainee/home/src/main/java/co/kr/tnt/trainee/home/TraineeHomeContract.kt:
```diff
@@ -1,21 +1,26 @@
-        val markedDates: List<LocalDate> = emptyList(),
-        val recordList: List<RecordList> = emptyList(),
+        val selectedDay: LocalDate = LocalDate.now(),
+        val dailyDataState: List<LocalDate> = emptyList(),
```
요 상태가 어떤 역할을 하는지 이해하지 못했어요 ㅜ

SeonJeongk replied:
`selectedDay`는 현재 달력에서 선택된 날짜이고,
`dailyDataState`는 달력에 PT 수업 혹은 기록이 있으면 icon을 표시하기 위해 해당하는 날짜들을 담고있습니다..!

hoyahozz commented on feature/trainee/home/src/main/java/co/kr/tnt/trainee/home/TraineeHomeScreen.kt:
```diff
@@ -62,82 +95,285 @@ internal fun TraineeHomeRoute(
     }
 }
 
+// 시간을 "오후 14:00"의 형식으로 바꿔준다
+@Composable
+private fun formatTime(isoString: String): String {
+    val timePart = isoString.substringAfter('T').substring(0, 5)
+    val hour = timePart.substring(0, 2).toInt()
+
+    val amPm = if (hour < 12) {
+        stringResource(uiResource.string.morning)
+    } else {
+        stringResource(uiResource.string.afternoon)
+    }
+
+    return "$amPm $timePart"
+}
+
+// 선택된 날짜를 "0월 0일 0요일"의 형식으로 바꿔준다
+private fun formatDateWithDay(date: LocalDate): String {
+    val monthDayFormatter = DateTimeFormatter.ofPattern("M월 d일", Locale.KOREAN)
+    val monthDay = date.format(monthDayFormatter)
+
+    val dayOfWeek = date.dayOfWeek.getDisplayName(TextStyle.FULL, Locale.KOREAN)
+
+    return "$monthDay $dayOfWeek"
+}
```
지금도 `DateFormatter` 를 사용할 수 있을 것처럼 보이는데, 혹시 API 연결 후에 사용해야 하는 이유가 있을까용 ?.?

SeonJeongk replied:
`TNT-179` branch 머지되면 해당 코드 기반으로 구현하려고 했습니다!!
언제 머지될지 모르겠어서 일단 뒤로 미뤘습니다..ㅎ

hoyahozz replied:
아항 머지해두었습니다!

SeonJeongk replied:
그렇담 수정해오겠습니다 🏃

SeonJeongk committed:
[TNT-181] fix: 기록 데이터 null 허용 제거

SeonJeongk committed:
[TNT-181] fix: 주간 이동 시 선택된 날짜 변경되지 않도록 수정

SeonJeongk committed:
[TNT-181] feat: develop branch DataFormatter 코드 추가

SeonJeongk committed:
[TNT-181] feat: 데이터 클래스 리팩토링 및 도메인 변환 구현

- 데이터 클래스 이름 변경 및 Response 데이터 클래스 추가
- Response 데이터를 도메인 모델로 변환
- dateFormatter를 활용해 날짜 및 시간 변환 처리

SeonJeongk committed:
[TNT-181] refactor: 변수명 변경

hoyahozz reviewed (approved):
고생하셨습니당 ~~~~!!! 👍👍👍
